game/game.py

```diff
@@ -386,26 +385,50 @@ def set_bullseye(self) -> None:
         self.blue_bullseye = Bullseye(enemy_cp.position)
         self.red_bullseye = Bullseye(player_cp.position)
 
-    def initialize_turn(self) -> None:
+    def initialize_turn(
+        self, for_red: bool = True, for_blue: bool = True, new_turn: bool = True
+    ):
         self.events = []
         self._generate_events()
-
         self.set_bullseye()
 
         # Update statistics
         self.game_stats.update(self)
 
-        self.blue_air_wing.reset()
-        self.red_air_wing.reset()
-        self.aircraft_inventory.reset()
-        for cp in self.theater.controlpoints:
-            self.aircraft_inventory.set_from_control_point(cp)
-
         # Check for win or loss condition
         turn_state = self.check_win_loss()
         if turn_state in (TurnState.LOSS, TurnState.WIN):
             return self.process_win_loss(turn_state)
 
+        # Plan Coalition specific turn
+        if for_red:
+            self.initialize_turn_for(player=False)
+        if for_blue:
+            self.initialize_turn_for(player=True)
+
+        # Plan GroundWar
+        for cp in self.theater.controlpoints:
+            if cp.has_frontline:
+                gplanner = GroundPlanner(cp, self)
+                gplanner.plan_groundwar()
+                self.ground_planners[cp.id] = gplanner
+
+        # Only plan procurement on a new turn
+        if new_turn:
+            if for_blue:
+                self.plan_procurement(True)
```

Comment:
And a minor thing: literal arguments should be passed as kwargs for clarity. `plan_procurement(player=True)` takes the guesswork out of what the argument is.

Reply:
totally agree on this one, i tried to do this one everywhere but i missed it here 🙄